target/linux/generic/pending-5.10/999-15-net-dsa-qca8k-cache-lo-and-hi-for-mdio-write.patch

```diff
@@ -0,0 +1,86 @@
+From 9419b504fd1119a2f0c3fcce1b67b8e5c437d62e Mon Sep 17 00:00:00 2001
+From: Ansuel Smith <ansuelsmth@gmail.com>
+Date: Tue, 7 Dec 2021 04:43:06 +0100
+Subject: [net-next RFC PATCH v4 15/15] net: dsa: qca8k: cache lo and hi for
+ mdio write
+
+From Documentation, we can cache lo and hi the same way we do with the
+page. This massively reduce the mdio write as 3/4 of the time we only
+require to write the lo or hi part for a mdio write.
+
+Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>
+---
+ drivers/net/dsa/qca8k.c | 49 ++++++++++++++++++++++++++++++++++++-----
+ 1 file changed, 44 insertions(+), 5 deletions(-)
+
+diff --git a/drivers/net/dsa/qca8k.c b/drivers/net/dsa/qca8k.c
+index 375a1d34e46f..b109a74031c6 100644
+--- a/drivers/net/dsa/qca8k.c
++++ b/drivers/net/dsa/qca8k.c
+@@ -94,6 +94,48 @@ qca8k_split_addr(u32 regaddr, u16 *r1, u16 *r2, u16 *page)
+ 	*page = regaddr & 0x3ff;
+ }
+ 
++static u16 qca8k_current_lo = 0xffff;
```

Comment:
qca8k_current_lo/high/page must be part of private data in structure qca8k_priv. If multiple qca8k switches does exist, currently they will share this three cached values.

Reply:
Aside from the the global values that just got fixed, (still needs to be merged but i'm spending some time clearing the pr)

Caching the values looks wrong IMHO and this caching is on the regs level so we don't cache values, we cache the indirect regs used for mdio.